pkg/planner/core/plan_cache.go

```diff
@@ -432,16 +437,28 @@ func rebuildRange(p Plan) error {
 			return err
 		}
 	case *PhysicalTableReader:
+		err = expression.ReEvaluateParamMarkerConst(sctx, x.PlanPartInfo.PruningConds)
+		if err != nil {
+			return err
+		}
 		err = rebuildRange(x.TablePlans[0])
 		if err != nil {
 			return err
 		}
 	case *PhysicalIndexReader:
+		err = expression.ReEvaluateParamMarkerConst(sctx, x.PlanPartInfo.PruningConds)
```

Comment:
Do we need to call `executor.partitionPruning` here, and update `PhysicalIndexReader.partitions`?